Add more explanation to to_time about timezones

kmayer opened the pull request:
There's a difference in the results between `Date#to_time(:local)` and `Date#in_time_zone` but it is subtle and can confuse users (like me :-).

`Date#to_time(:local)` will convert the date to Ruby's process timezone, i.e. `ENV['TZ']` but when working in a Rails application, you generally want to use the *application's* timezone, i.e. `Time.zone`. There's a method for that: `Date#in_time_zone`, but it's off in another file (https://github.com/rails/rails/blob/92703a9ea5d8b96f30e0b706b801c9185ef14f0e/activesupport/lib/active_support/core_ext/date_and_time/zones.rb)

maclover7 reviewed (changes requested):
Can you squash down your commits to one? After that looks ready to me :)

kmayer committed:
Add (more) documentation to to_time

There's a difference in the results between `Date#to_time(:local)` and
`Date#in_time_zone` but it is subtle and can confuse users (like me :-).

kmayer commented:
@maclover7 squished

maclover7 commented:
:+1: Thank you for your contribution!
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
